pandas/core/generic.py

```diff
@@ -3747,7 +3750,10 @@ class   animal   locomotion
                 name=self.index[loc],
                 dtype=new_values.dtype,
             )
-
+        elif is_scalar(loc):
+            result = self.iloc[:, [loc]]
```

Comment:
Until we implement #36195, this returns a copy and not a view.  In followup can you make this `self.iloc[:, slice(loc, loc + 1)]`, probably need to test that we're getting a view

Reply:
#37832